frontend/src/TakerApp.tsx

```diff
@@ -115,6 +115,24 @@ export default function App() {
     const open = cfds.filter((value) => value.state.getGroup() === StateGroupKey.OPEN);
     const closed = cfds.filter((value) => value.state.getGroup() === StateGroupKey.CLOSED);
 
+    if (quantity == "0" && order?.min_quantity) {
+        console.log("Updating quantity");
+        updateQuantityAndMargin(order?.min_quantity.toString());
+    }
+    function updateQuantityAndMargin(valueString: string) {
+        setQuantity(parse(valueString));
+        if (!order) {
+            return;
+        }
+        let quantity = valueString ? Number.parseFloat(valueString) : 0;
+        let payload: MarginRequestPayload = {
+            leverage: order.leverage,
+            price: order.price,
+            quantity,
+        };
+        calculateMargin(payload);
+    }
```

Comment:
Side effects should always happen within `useEffect`!

Reply:
Hang on, this is not a side-effect at all. This should be changed to just use derivation.

Reply:
> This should be changed to just use derivation.

Can you clarify please, I don't follow :)

Reply:
oh, thanks for the clear explanation. 

That means I cannot update the `margin` unless I would calculate it here as well.

I'll leave the default value of `quantity` to `0` then. This will make the field invalid as `0` won't be within the bounds but I think that is better than having `margin=0` and `quantity=100`.

Reply:
> That means I cannot update the `margin` unless I would calculate it here as well.

You can, you just need to do it in a controlled way via `useEffect` where you declare all dependencies upon which you wish to cause a re-render.

Reply:
Alternatively, and that is often the better solution, these kind of situations make it visible that there is actually state we care about, but are not tracking at the moment, for example whether or not the use has edited the field already. We had a similar situation in the past in regards to calculating the initial price.

Reply:
I'm interested in the second option. What do you have in mind? Indicating that the user hasn't edited the field and don't show the margin?

Reply:
If I understand correctly, what you are trying to achieve is to set a useful default value yes? To model a default value, we have to keep track of the state where the value came from (initial vs user-filled in). So we can do something like:

```ts
let [userHasEdited, setUserHasEdited] = useState(false);	
let quantityToShow = userHasEdited ? quantity : order.minQuantity;
```

And do `setUserHasEdited(true)` in the onChange handler of the input field.

Reply:
Like this [`8450ef6` (#307)](https://github.com/comit-network/hermes/pull/307/commits/8450ef6fb0cb50397590a3fa843b4a6563e20f55)?